### PR TITLE
feat: update unix-shell-evasion-prefix

### PR DESCRIPTION
**Pull request by Xhoenix**
fixes JX9-231209-1

### CODE REVIEW

File: regex-assembly/include/unix-shell-evasion-prefix.ra [hunk 1]
```diff
@@ -40,6 +41,8 @@
 >\(
 ##! a() ( ifconfig; ); a
 \(\s*\)
+##! `cat<<<ifconfig` or `cat<<< ifconfig`
+<<<
```
Comment (theseion) [inline regex-assembly/include/unix-shell-evasion-prefix.ra:45]: You might be missing optional white space here, not sure. Please add a couple of tests as this affects about 6 rules.

**Review by theseion — CHANGES_REQUESTED** [review]

File: tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml [hunk 2]
```diff
@@ -216,3 +216,39 @@ tests:
             version: HTTP/1.1
           output:
             log_contains: id "932235"
+  - test_title: 932235-13
+    desc: "JX9-231209-1"
```
Comment (theseion) [inline tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:220]:
```suggestion
    desc: "Block evasion using here string"
```

File: tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml [hunk 3]
```diff
@@ -216,3 +216,39 @@ tests:
...
+            method: POST
+            port: 80
+            uri: /post
+            version: HTTP/1.0
+            data: |
+              code=;%60cat<<<ifconfig%60
+          output:
+            log_contains: id "932235"
+  - test_title: 932235-14
+    desc: "JX9-231209-1 with a space"
```
Comment (theseion) [inline tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:238]:
```suggestion
    desc: "Block evasion using here string with space"
```

File: regex-assembly/include/unix-shell-evasion-prefix.ra [hunk 4]
```diff
@@ -6,9 +6,15 @@
 ##! Separate rules target commands that do not follow this prefix,
 ##! as the chance of false positives is higher without a prefix match.
 
-##! time ifconfig
+##! time ifconfig and watch ifconfig
```
Comment (theseion) [inline regex-assembly/include/unix-shell-evasion-prefix.ra:9]:
```suggestion
##! <some command> ifconfig
```

File: tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml
Comment (theseion) [inline tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:1]: Here strings can also begin with single quote, double quote or dollar (for expansion or process substitution). Please add tests for those as well.

Comment (Xhoenix) [reply tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:1]: As per bash documentation, I've payloads all bash "expansions" and command substitution. Working on a "quote removal" payload...

Comment (theseion) [reply tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:1]: Not sure what you mean. I was thinking of these:
```shell
cat<<<'foo'
cat<<<"foo"
cat<<<"${foo}"
cat<<<${foo}
cat<<<"$foo"
cat<<<$foo
cat<<<"$(echo foo)"
cat<<<$(echo foo)
```

Comment (Xhoenix) [reply tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:1]: The quotes still need backticks to work, but I guess you meant as a part of the original payload.

Comment (theseion) [reply tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:1]: Are done yet? Should I do another review pass?

Comment (Xhoenix) [reply tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:1]: Okay, LGTM!

Comment (theseion) [inline tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:1]: Please duplicate the new tests for 932230, 932231, 932220, 932236, 932238, 932239. All of them are affected by the addition of here strings.

Please also add at least one test (for each rule) that tests the new additions (e.g., `strace`).

Comment (Xhoenix) [reply tests/regression/tests/REQUEST-932-APPLICATION-ATTACK-RCE/932235.yaml:1]: I don't think they're required @theseion. Kindly make any changes if you think necessary.

**Comment by theseion** [comment]
I messed up the force push. I've opened a new PR with my additions: #3518.

**Comment by Xhoenix** [comment]
By "if you think necessary", i didn't mean take my PR. 😅

**Pull request closed without merging**